### ISSUE TITLE
Featured product "tax_query" argument overwritten

### ISSUE THREAD

**Issue by lstellway** [wp-graphql/wp-graphql-woocommerce#370]
**Describe the bug**
When querying "featured" products in a category product connection, the `tax_query` argument is overwritten. 

**To Reproduce**
1. Setup WooCommerce store with products and WPGraphQL extensions
2. Mark a single item as featured
3. Query the category products, with a `featured` filter
```
query GetProducts {
  productCategory(idType:SLUG, id:"uncategorized") {
    products(where: {
      featured:true
    }) {
      nodes {
        name
      }
    }
  }
}
```

**Expected behavior**
Only the featured item should appear.

**Desktop (please complete the following information):**
 - OS: OSX
 - Browser: Postman

**Additional context**
The issue occurs somewhere between [class-products.php::$resolve_product_from_taxonomy()](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/ddf7661ccfe5d883a155611a85b4fec7dbb7b909/includes/connection/class-products.php#L231-L238) and [class-product-connection-resolver.php::set_query_arg()](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/664bacf42969ce189b5a8eebce086b1881b08517/includes/data/connection/class-product-connection-resolver.php#L585-L593).

I'm curious if the logic in [class-product-connection-resolver.php::set_query_arg()](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/664bacf42969ce189b5a8eebce086b1881b08517/includes/data/connection/class-product-connection-resolver.php#L585-L593) is correct.
Should the `$overwrite` variable in the `if` statement have a `!` (bang) in front of it? 
I would think so, so the `tax_query` is appended to instead of overwritten.

**Comment by loganstellway** [comment]
It sounds like this logic is going to be refactored ([reference](https://github.com/wp-graphql/wp-graphql-woocommerce/pull/371#pullrequestreview-532877663)). 

Here's a patch in case anyone is running into the same issue and wants a temporary fix -
_(if you're using composer, you can include it with [cweagans/composer-patches](https://github.com/cweagans/composer-patches))_
```
diff --git a/includes/data/connection/class-product-connection-resolver.php b/includes/data/connection/class-product-connection-resolver.php
index 97fcda2..bec3ad5 100644
--- a/includes/data/connection/class-product-connection-resolver.php
+++ b/includes/data/connection/class-product-connection-resolver.php
@@ -584,9 +584,9 @@ class Product_Connection_Resolver extends AbstractConnectionResolver {
 	 */
 	public function set_query_arg( $key, $value, $overwrite = false ) {
 		if ( ! empty( $this->query_args[ $key ] )
-			&& is_array( $this->query_args[ $key ] ) && $overwrite
+			&& is_array( $this->query_args[ $key ] ) && ! $overwrite
 		) {
-			$this->query_args[ $key ] = array_intersect( $value, $this->query_args[ $key ] );
+			$this->query_args[ $key ][] = $value;
 		} else {
 			$this->query_args[ $key ] = $value;
 		}
```

**Comment by loganstellway** [comment]
I am experiencing this again -
_(my patch became obsolete after updating)_. 

The [`tax_query` argument is set](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/ddf7661ccfe5d883a155611a85b4fec7dbb7b909/includes/connection/class-products.php#L231-L238) when products are resolved. When stepping through the code with `xdebug`, the referenced statement was used to filter the category. 

Because of this statement, I cannot [filter "featured" products](https://github.com/wp-graphql/wp-graphql-woocommerce/blob/bc07902b35964822205b8e22cfcd4c154d67010b/includes/data/connection/class-product-connection-resolver.php#L383-L405), which relies on the `tax_query` parameter. 

Removing the old `WPGraphQL\WooCommerce\Data\Connection\Product_Connection_Resolver::set_query_arg()` method causes the entire `tax_query` argument to be overwritten. 

To fix for myself, I have created another patch, placing the old function back:
```
diff --git a/includes/data/connection/class-product-connection-resolver.php b/includes/data/connection/class-product-connection-resolver.php
index 3322830..4f4ecf1 100644
--- a/includes/data/connection/class-product-connection-resolver.php
+++ b/includes/data/connection/class-product-connection-resolver.php
@@ -569,4 +569,23 @@ class Product_Connection_Resolver extends AbstractConnectionResolver {
 	public function is_valid_offset( $offset ) {
 		return $this->is_valid_post_offset( $offset );
 	}
+
+	/**
+	 * Given a key and value, this sets a query_arg which will modify the query_args used by
+	 * the connection resolvers get_query();
+	 *
+	 * @param string $key   The key of the query arg to set
+	 * @param mixed  $value The value of the query arg to set
+	 *
+	 * @return AbstractConnectionResolver
+	 */
+	public function set_query_arg( $key, $value, $overwrite = false ) {
+		if ( is_array( $this->query_args ) && ! empty( $this->query_args[ $key ] ) && ! $overwrite ) {
+			$this->query_args[ $key ][] = $value;
+		} else {
+			$this->query_args[ $key ] = $value;
+		}
+
+		return $this;
+	}
 }
```

**Comment by kidunot89** [comment]
This functionality was finally fixed by the merging of #820
